### PR TITLE
Adding support for MapData types

### PR DESCRIPTION
**Pull request by ohadbitt**
Adding support for MapData types

### CODE REVIEW

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala [hunk 1]
```diff
@@ -335,43 +340,51 @@ object KustoWriter {
     if (row.get(fieldIndexInRow, dataType) == null) null else dataType match {
       case DateType => DateTimeUtils.toJavaDate(row.getInt(fieldIndexInRow)).toString
       case TimestampType => dateFormat.format(DateTimeUtils.toJavaTimestamp(row.getLong(fieldIndexInRow)))
-      case StringType => GetStringFromUTF8(row, fieldIndexInRow, nested)
+      case StringType => GetStringFromUTF8(row.getUTF8String(fieldIndexInRow), nested)
       case BooleanType => row.getBoolean(fieldIndexInRow).toString
-      case _: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), dataType.asInstanceOf[StructType], dateFormat)
-      case _: ArrayType => convertArrayToCsv(row.getArray(fieldIndexInRow), dataType, dateFormat)
+      case strType: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), strType, dateFormat)
```
Comment (toshetah) [inline connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala]: Why strType and not _ ? Don't you get a warning of an unused variable?

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala [hunk 2]
```diff
@@ -335,43 +340,51 @@ object KustoWriter {
...
+    val result: Array[String] = new Array(ar.numElements())
+    for (x <- 0 until ar.numElements()) {
+      result(x) = getField(ar, x, fieldsType, dateFormat, nested)
+    }
+    result.mkString(",")
+  }
+
+
+  private def convertMapToCsv(map: MapData, fieldsType: MapType, dateFormat: FastDateFormat): String = {
+    val a = convertArrayToCsvImpl(map.keyArray(), fieldsType.keyType, dateFormat, nested = false)
```
Comment (toshetah) [inline connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala]: Please give significant names, and not a, and b.

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala [hunk 3]
```diff
@@ -335,43 +340,51 @@ object KustoWriter {
...
+    val b = convertArrayToCsvImpl(map.valueArray(), fieldsType.valueType, dateFormat)
+    "{" +"\"" + a + "\"" + ":"+ b + "}"
```
Comment (toshetah) [inline connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala]: This logic is not correct.
Let's say we have map.keyArray() = ["a","b"] and map.valueArray() = [1,2]
So we get:
a = "a","b" 
b = 1,2
and the result of this method is:
{""a","b" ":1,2}

What we want is:
{"a":1,"b":2}

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala (hunk 1 above)
Comment (ohadbitt) [reply connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala]: i use it .. this replaces the calling to asInstanceOf[StructType]

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala [hunk 4]
```diff
@@ -335,43 +340,58 @@ object KustoWriter {
...
+        "[" + convertArrayToCsvImpl(ar, fieldsType, dateFormat).mkString(",") + "]"
       }
     }
   }
 
-  private def GetStringFromUTF8(ar: SpecializedGetters, index: Int, nested: Boolean) = {
-    val str = ar.getUTF8String(index)
-    if (str == null) null else
-      if (nested) "\"" + str.toString + "\"" else str.toString
+  private def convertArrayToCsvImpl(ar: ArrayData, fieldsType: DataType, dateFormat: FastDateFormat, nested: Boolean = true): Array[String] = {
```
Comment (toshetah) [inline connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala]: convertArrayToStringArray

File: connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala [hunk 5]
```diff
@@ -335,43 +340,58 @@ object KustoWriter {
     if (row.get(fieldIndexInRow, dataType) == null) null else dataType match {
       case DateType => DateTimeUtils.toJavaDate(row.getInt(fieldIndexInRow)).toString
       case TimestampType => dateFormat.format(DateTimeUtils.toJavaTimestamp(row.getLong(fieldIndexInRow)))
-      case StringType => GetStringFromUTF8(row, fieldIndexInRow, nested)
+      case StringType => GetStringFromUTF8(row.getUTF8String(fieldIndexInRow), nested)
       case BooleanType => row.getBoolean(fieldIndexInRow).toString
-      case _: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), dataType.asInstanceOf[StructType], dateFormat)
-      case _: ArrayType => convertArrayToCsv(row.getArray(fieldIndexInRow), dataType, dateFormat)
+      case structType: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), structType, dateFormat)
```
Comment (toshetah) [inline connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala]:
```suggestion
      case structType: StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, structType.length), structType, dateFormat)
```

**Pull request merged**